### DISCUSSION TITLE
Dataset Tabularization and In-sample Prediction

### DISCUSSION THREAD

**Discussion by anthonygiorgio97** [alan-turing-institute/sktime#2569] [Q&A]
Hi,
I was wondering if it is possible to get the tabularized version of a time series specifying a window length (as obtained internally from `make_reduction()` function). Furthermore, it is possible to obtain in sample predictions using  `make_reduction()` and sklearn algorithms? This is possible with AutoARIMA, Prophet by specifing the train index for the forecasting horizon in the `.predict()` method . But when I try this with a ML method using `make_reduction()` it raises an error.

**Comment by fkiraly** [comment]
> I was wondering if it is possible to get the tabularized version of a time series specifying a window length

You can do this currently with `WindowSummarizer`, although you have to give it a `range` instead of directly specifying window length.
There is some work underway to make this available under a simpler interface, a lagging transformer: https://github.com/alan-turing-institute/sktime/issues/2526

>Furthermore, it is possible to obtain in sample predictions using make_reduction() and sklearn algorithms? This is possible with AutoARIMA

That does not seem possible at the moment - thank you for alerting me to this, there was no issue that kept track that this was still something to do. I´ve opened one now:
https://github.com/alan-turing-institute/sktime/issues/2591
